src/codons.py

```diff
@@ -40,6 +45,13 @@ def split_codons(dna: str) -> list[str] | None:
 
 
 def translate_codons(codons: list[str]) -> list[str]:
+    amino = []
+    for i in codons:
+        if i.upper() not in CODON_MAP:
+            return None
+        else i.upper() in CODON_MAP:
```

Comment:
Here you are in the same situation with returning from the `if` block. You don't need to check for the complement.